tests/test_models.py

```diff
@@ -150,6 +150,28 @@ def test_create_a_supplier_with_no_name(self):
         """ Create a Suppleir with no name """
         supplier = Supplier(None, 2, True, [1,2,3], 8.5)
         self.assertRaises(DataValidationError, supplier.create)
+    
+    def test_find_supplier(self):
+        """ Find a Supplier by id """
+        Supplier("supplier1", 2, True, [1,2,3], 8.5).save()
+        # saved_supplier = Supplier("kitty", "cat").save()
+        saved_supplier = Supplier("supplier1", 2, True, [1,2,3], 8.5)
+        saved_supplier.save()
+        supplier = Supplier.find(saved_supplier.id)
+        self.assertIsNot(supplier, None)
+        self.assertEqual(supplier.id, saved_supplier.id)
+        self.assertEqual(supplier.name, "supplier1")
+    
+    def test_find_with_no_suppliers(self):
+        """ Find a Supplier with empty database """
+        supplier = Supplier.find("1")
+        self.assertIs(supplier, None)
+
+    def test_supplier_not_found(self):
```

Comment:
It seems that the test case ```test_supplier_not_found``` is actually the same as ```test_find_with_no_suppliers```. Maybe we can just keep ```test_supplier_not_found```. Better to name as "test_find_supplier_no_result"

Reply:
The are similar, but "test_supplier_not_found" is a test for func in service.py, "test_find_with_no_suppliers" is a test for func in model.py. They cover different parts.